bundles/http/Citrix/Workday-Demo/script.js

```diff
@@ -0,0 +1,192 @@
+/*
+Name: Workday Demo HTTP integration, using generated dummy data
+Created by: Andrey Kornilov
+Date: 18.7.2021
+*/
+
+const _ = library.load('lodash')
+
+const names = ["Aline Gomes", "Billy Taylor", "Danielle Ledford", "Diana Sarkozy", "Erberto Tirado"]
+const descriptions = ["Lyft","Uber","Dinner","Hotel","Flight Tickets","Rental"]
+
+//Function, that generates data for the table, using random values, both for incremental and full sync
+async function getExpenses (dataStore,context,fullSync) {
+  let iterations;
+  let dateRandom;
+  const min = 100
+  const max = 2000
+//Switch statement, that regulates behavior of loop, depending on sync type
+  switch (fullSync) {
+    case true:
+      context.key1 = 1;
+      iterations = context.key1 + 15
+      dateRandom = true
+      break;
+
+    default:
+      iterations = context.key1 + 3
+      dateRandom = false
+      break;
+  }
+const expenseReport = _.range(context.key1, iterations).map(ID => {
+			const min = 100
+      const max = 20000
+      //Init date instance and check, if we need to randomize dates
+      let date = new Date()
+      dateRandom ? date.setDate(date.getDate()-(Math.floor(Math.random() * (10)))) : date.setDate(date.getDate());
```

Comment:
`const date = moment().subtract(Math.random() * 10, 'days').toISOString()` would be more convenient